pkg/localstore/subscription_pull.go

```diff
@@ -36,7 +36,7 @@ import (
 // function will terminate current and further iterations without errors, and also close the returned channel.
 // Make sure that you check the second returned parameter from the channel to stop iteration when its value
 // is false.
-func (db *DB) SubscribePull(ctx context.Context, bin uint8, since, until uint64) (c <-chan storage.Descriptor, stop func()) {
+func (db *DB) SubscribePull(ctx context.Context, bin uint8, since, until uint64) (c <-chan storage.Descriptor, closed <-chan struct{}, stop func()) {
```

Comment:
theoretically yes. in the old codebase for some reason though this wasn't so for some reason and you could get some empty batch on shutdown. the problem is also that `SubscribePull` does not provide an explicit signal about shutdown (before this change). I rather be on the safe side for now. But theoretically yes you are right and it could be that the entire section of code here which is handling shutdown could be removed :+1: Need to test this though to verify the assumption.